File: python/0069-e-sqrtx.py
# https://leetcode.com/problems/sqrtx
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0: return 0
        left = 1
        right = x

        while left <= right:
            mid = left + (right - left) // 2
            times = mid * mid
            if times == x:
                return mid
            elif times > x:
                right = mid - 1
            else:
                left = mid + 1
        
        return left - 1

File: kotlin/0069-e-sqrtx.kt
class Solution {
    fun mySqrt(x: Int): Int {
        if (x == 0) return 0

        var left = 1
        var right = x

        while (left <= right) {
            val mid = left + (right - left) / 2
            val divide = (x/mid).toInt()
            if (divide == mid) return mid
            else if (divide < mid) right = mid - 1
            else left = mid + 1
        }

        return left - 1
    }
}